### PR TITLE
Publish-Subscribe bug fix

### PR DESCRIPTION
**Pull request by RaulinN**
Subscribe and unsubscribe requests were basically ignored by the mediator. This can be done since the ask pattern is now working in the system (PR #411)

### CODE REVIEW

File: be2-scala/src/main/scala/ch/epfl/pop/pubsub/PubSubMediator.scala [hunk 1]
```diff
@@ -14,6 +14,7 @@ class PubSubMediator extends Actor with ActorLogging {
   private var channelMap: mutable.Map[Channel, mutable.Set[ActorRef]] = mutable.Map.empty
 
   override def receive: Receive = LoggingReceive {
+    // FIXME pubsub mediator doesnt check that the channel exists in db
```
Comment (pierluca) [inline be2-scala/src/main/scala/ch/epfl/pop/pubsub/PubSubMediator.scala:17]: I'd fix this before merging

**Review by pierluca — COMMENTED** [review]
I don't see any obvious mistake, we'll have to test :)

File: be2-scala/src/main/scala/ch/epfl/pop/pubsub/graph/handlers/ParamsHandler.scala [hunk 2]
```diff
@@ -50,23 +56,60 @@ object ParamsHandler {
...
+    case Left(jsonRpcMessage: JsonRpcResponse) =>
+      Right(PipelineError(ErrorCodes.SERVER_ERROR.id, "UnsubscribeHandler received a 'JsonRpcResponse'", jsonRpcMessage.id))
     case graphMessage@_ => graphMessage
   }
 
-  def catchupHandler(clientActorRef: ActorRef): Flow[GraphMessage, GraphMessage, NotUsed] = Flow[GraphMessage].map {
+  def catchupHandler(clientActorRef: AskableActorRef): Flow[GraphMessage, GraphMessage, NotUsed] = Flow[GraphMessage].map {
     // case Left(jsonRpcMessage: Catchup) => clientActorRef ! ClientActor.CatchupChannel(jsonRpcMessage.channel)
-    case graphMessage@_ => graphMessage // FIXME catchup
+    case graphMessage@_ => graphMessage // FIXME catchup handler
```
Comment (pierluca) [inline be2-scala/src/main/scala/ch/epfl/pop/pubsub/graph/handlers/ParamsHandler.scala:113]: shouldn't this be addressed too?

Comment (RaulinN) [reply be2-scala/src/main/scala/ch/epfl/pop/pubsub/graph/handlers/ParamsHandler.scala:113]: Resolved #449

File: be2-scala/src/main/scala/ch/epfl/pop/pubsub/PubSubMediator.scala (hunk 1 above)
Comment (pierluca) [reply be2-scala/src/main/scala/ch/epfl/pop/pubsub/PubSubMediator.scala:17]: Is this resolved in the other PR too? 
If so, feel free to merge.

**Review by pierluca — DISMISSED** [review]
Merge if you think it's appropriate at this point.

File: be2-scala/src/main/scala/ch/epfl/pop/pubsub/PubSubMediator.scala (hunk 1 above)
Comment (RaulinN) [reply be2-scala/src/main/scala/ch/epfl/pop/pubsub/PubSubMediator.scala:17]: Resolved #449

**Review by pierluca — APPROVED** [review]

**Pull request merged**